project_redss/auto_code_surveys.py

```diff
@@ -48,17 +48,21 @@ def auto_code_surveys(user, data, phone_uuid_table, coda_output_dir):
                     td.append_data({"district_coded": nc_label.to_dict()},
                                    Metadata(user, Metadata.get_call_location(), time.time()))
 
-        # TODO: Auto-code operator + channels
-        # # Label each message with the operator of the sender
-        # for td in data:
-        #     phone_number = phone_uuid_table.get_phone(td["avf_phone_id"])
-        #     operator = PhoneCleaner.clean_operator(phone_number)
-        #
-        #     td.append_data(
-        #         {"operator": operator},
-        #         Metadata(user, Metadata.get_call_location(), time.time())
-        #     )
-        #
+        # Set operator from phone number
+        for td in data:
+            operator_clean = PhoneCleaner.clean_operator(phone_uuid_table.get_phone(td["uid"]))
+            if operator_clean == Codes.NOT_CODED:
```

Comment:
Please can you monitor as to how often this pattern of needing to switch between CodeID and Control code occurs, and I might revisit the decision to not put matching for Control codes. A few instances isn't sufficient to justify but if it becomes very common boiler plate we might look again.

Reply:
I think this is the only time I've seen it. It's certainly not come up a lot